### DISCUSSION TITLE
What is the correct version of dropwizard-archtype?

### DISCUSSION THREAD

**Discussion by sulman-hungryepos** [dropwizard/dropwizard#4760] [Q&A] [answered]
When I try to create a new project using 

`mvn archetype:generate -DarchetypeGroupId=io.dropwizard.archetypes -DarchetypeArtifactId=java-simple -DarchetypeVersion=2.1.0-SNAPSHOT
`
I get following:

```
 BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.611 s
[INFO] Finished at: 2020-07-18T19:13:29+10:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-archetype-plugin:3.1.2:generate (default-cli) on project standalone-pom: The desired archetype does not exist (io.dropwizard.archetypes:java-simple:2.1.0-SNAPSHOT) -> [Help 1]

```

**Comment by joschi** [comment, marked as answer]
The Maven Archetype is published together with each Dropwizard release, so using `io.dropwizard.archetypes:java-simple:2.0.11` should do the trick unless you want to test something with the latest SNAPSHOT version (in which case you have to build it yourself first).

```text
% mvn archetype:generate -DarchetypeGroupId=io.dropwizard.archetypes -DarchetypeArtifactId=java-simple -DarchetypeVersion=2.0.11
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------< org.apache.maven:standalone-pom >-------------------
[INFO] Building Maven Stub Project (No POM) 1
[INFO] --------------------------------[ pom ]---------------------------------
[INFO]
[INFO] >>> maven-archetype-plugin:3.1.2:generate (default-cli) > generate-sources @ standalone-pom >>>
[INFO]
[INFO] <<< maven-archetype-plugin:3.1.2:generate (default-cli) < generate-sources @ standalone-pom <<<
[INFO]
[INFO]
[INFO] --- maven-archetype-plugin:3.1.2:generate (default-cli) @ standalone-pom ---
[INFO] Generating project in Interactive mode
[INFO] Archetype repository not defined. Using the one from [io.dropwizard.archetypes:java-simple:2.0.11] found in catalog remote
Downloading from central: https://repo1.maven.org/maven2/io/dropwizard/archetypes/java-simple/2.0.11/java-simple-2.0.11.pom
Downloaded from central: https://repo1.maven.org/maven2/io/dropwizard/archetypes/java-simple/2.0.11/java-simple-2.0.11.pom (1.5 kB at 10 kB/s)
Downloading from central: https://repo1.maven.org/maven2/io/dropwizard/archetypes/dropwizard-archetypes/2.0.11/dropwizard-archetypes-2.0.11.pom
Downloaded from central: https://repo1.maven.org/maven2/io/dropwizard/archetypes/dropwizard-archetypes/2.0.11/dropwizard-archetypes-2.0.11.pom (8.1 kB at 124 kB/s)
Downloading from central: https://repo1.maven.org/maven2/io/dropwizard/archetypes/java-simple/2.0.11/java-simple-2.0.11.jar
Downloaded from central: https://repo1.maven.org/maven2/io/dropwizard/archetypes/java-simple/2.0.11/java-simple-2.0.11.jar (3.9 kB at 29 kB/s)
Define value for property 'groupId': [...]
```

**Reply by DavidNovo** [reply]
Hello, I just tried your command on my system:
% mvn archetype:generate -DarchetypeGroupId=io.dropwizard.archetypes -DarchetypeArtifactId=java-simple -DarchetypeVersion=2.0.11

I get this error stack:

```
> [INFO] Scanning for projects...
> [INFO]
> [INFO] ------------------< org.apache.maven:standalone-pom >-------------------
> [INFO] Building Maven Stub Project (No POM) 1
> [INFO] --------------------------------[ pom ]---------------------------------
> [INFO]
> [INFO] >>> maven-archetype-plugin:2.0-alpha-2:generate (default-cli) > generate-sources @ standalone-pom >>>
> [INFO]
> [INFO] <<< maven-archetype-plugin:2.0-alpha-2:generate (default-cli) < generate-sources @ standalone-pom <<<
> [INFO]
> [INFO]
> [INFO] --- maven-archetype-plugin:2.0-alpha-2:generate (default-cli) @ standalone-pom ---
> [INFO] Setting property: classpath.resource.loader.class => 'org.codehaus.plexus.velocity.ContextClassLoaderResourceLoader'.
> [INFO] Setting property: velocimacro.messages.on => 'false'.
> [INFO] Setting property: resource.loader => 'classpath'.
> [INFO] Setting property: resource.manager.logwhenfound => 'false'.
> [WARNING] No archetype repository found. Falling back to central repository (http://repo1.maven.org/maven2).
> [WARNING] Use -DarchetypeRepository=<your repository> if archetype's repository is elsewhere.
> org.apache.maven.archetype.downloader.DownloadException: Error downloading.
> 	at org.apache.maven.archetype.downloader.DefaultDownloader.download(DefaultDownloader.java:58)
> 	at org.apache.maven.archetype.common.DefaultArchetypeArtifactManager.exists(DefaultArchetypeArtifactManager.java:310)
> 	at org.apache.maven.archetype.ui.DefaultArchetypeGenerationConfigurator.configureArchetype(DefaultArchetypeGenerationConfigurator.java:103)
> 	at org.apache.maven.archetype.mojos.CreateProjectFromArchetypeMojo.execute(CreateProjectFromArchetypeMojo.java:168)
> 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
> 	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:370)
> 	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:351)
> 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:215)
> 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:171)
> 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:163)
> 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
> 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
> 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
> 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
> 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:298)
> 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
> 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
> 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:960)
> 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:293)
> 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:196)
> 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:577)
> 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
> 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
> 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
> 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
> Caused by: org.apache.maven.artifact.resolver.ArtifactResolutionException: Could not transfer artifact io.dropwizard.archetypes:java-simple:jar:2.0.11 from/to maven-default-http-blocker (http://0.0.0.0/): Blocked mirror for repositories: [java-simple-repo (http://repo1.maven.org/maven2, default, releases+snapshots)]
>   io.dropwizard.archetypes:java-simple:jar:2.0.11
> 
> from the specified remote repositories:
>   maven-default-http-blocker (http://0.0.0.0/, releases=true, snapshots=true)
> 
> 	at org.apache.maven.artifact.resolver.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:225)
> 	at org.apache.maven.artifact.resolver.DefaultArtifactResolver.resolveAlways(DefaultArtifactResolver.java:161)
> 	at org.apache.maven.archetype.downloader.DefaultDownloader.download(DefaultDownloader.java:54)
> 	... 25 more
> Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: Could not transfer artifact io.dropwizard.archetypes:java-simple:jar:2.0.11 from/to maven-default-http-blocker (http://0.0.0.0/): Blocked mirror for repositories: [java-simple-repo (http://repo1.maven.org/maven2, default, releases+snapshots)]
> 	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:425)
> 	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:229)
> 	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact(DefaultArtifactResolver.java:207)
> 	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveArtifact(DefaultRepositorySystem.java:262)
> 	at org.apache.maven.artifact.resolver.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:215)
> 	... 27 more
> Caused by: org.eclipse.aether.transfer.ArtifactTransferException: Could not transfer artifact io.dropwizard.archetypes:java-simple:jar:2.0.11 from/to maven-default-http-blocker (http://0.0.0.0/): Blocked mirror for repositories: [java-simple-repo (http://repo1.maven.org/maven2, default, releases+snapshots)]
> 	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads(DefaultArtifactResolver.java:521)
> 	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:402)
> 	... 31 more
> Caused by: org.eclipse.aether.transfer.NoRepositoryConnectorException: Blocked mirror for repositories: [java-simple-repo (http://repo1.maven.org/maven2, default, releases+snapshots)]
> 	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads(DefaultArtifactResolver.java:507)
> 	... 32 more
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD FAILURE
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time:  0.634 s
> [INFO] Finished at: 2023-01-18T20:24:47-06:00
> [INFO] ------------------------------------------------------------------------
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-archetype-plugin:2.0-alpha-2:generate (default-cli) on project standalone-pom: The desired archetype does not exist (io.dropwizard.archetypes:java-simple:2.0.11) -> [Help 1]
> [ERROR]
> [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
> [ERROR] Re-run Maven using the -X switch to enable full debug logging.
> [ERROR]
> [ERROR] For more information about the errors and possible solutions, please read the following articles:
> [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

Here is my version of Maven:
```
└─(20:27:26)──> whereis mvn                                                    
mvn: /usr/local/bin/mvn
└─(20:27:32)──> mvn -version                                                        
Apache Maven 3.8.7 (b89d5959fcde851dcb1c8946a785a163f14e1e29)
Maven home: /usr/local/Cellar/maven/3.8.7/libexec
Java version: 18.0.2, vendor: Amazon.com Inc., runtime: /Users/davidnovogrodsky_wrk/.sdkman/candidates/java/18.0.2-amzn
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "12.6.2", arch: "x86_64", family: "mac"
```

The file  /usr/local/bin/mvn is an alais that points to Maven home.

**Reply by joschi** [reply]
> ```
> Caused by: org.eclipse.aether.transfer.ArtifactTransferException: Could not transfer artifact io.dropwizard.archetypes:java-simple:jar:2.0.11 from/to maven-default-http-blocker (http://0.0.0.0/): Blocked mirror for repositories: [java-simple-repo (http://repo1.maven.org/maven2, default, releases+snapshots)]
> ```

Maven started blocking HTTP repositories in version 3.8.1, see https://maven.apache.org/docs/3.8.1/release-notes.html.

Check your Maven configuration for repositories accessed via HTTP (instead of HTTPS) and try using version 2.1.4 of the Dropwizard archtetype.

FYI: Dropwizard 2.0.x will reach EOL in 12 days on January 31, 2023.
https://github.com/dropwizard/dropwizard/discussions/6324

**Reply by DavidNovo** [reply]
Thank you.

**Reply by DavidNovo** [reply]
I tried version 2.1.4 as you suggested:

> mvn archetype:generate -DarchetypeGroupId=io.dropwizard.archetypes -DarchetypeArtifactId=java-simple -DarchetypeVersion=2.1.4

Unfortunately this was the error stack I received:

> [INFO] Scanning for projects...

> [INFO]
> [INFO] ------------------< org.apache.maven:standalone-pom >-------------------
> [INFO] Building Maven Stub Project (No POM) 1
> [INFO] --------------------------------[ pom ]---------------------------------
> [INFO]
> [INFO] >>> maven-archetype-plugin:2.0-alpha-2:generate (default-cli) > generate-sources @ standalone-pom >>>
> [INFO]
> [INFO] <<< maven-archetype-plugin:2.0-alpha-2:generate (default-cli) < generate-sources @ standalone-pom <<<
> [INFO]
> [INFO]
> [INFO] --- maven-archetype-plugin:2.0-alpha-2:generate (default-cli) @ standalone-pom ---
> [INFO] Setting property: classpath.resource.loader.class => 'org.codehaus.plexus.velocity.ContextClassLoaderResourceLoader'.
> [INFO] Setting property: velocimacro.messages.on => 'false'.
> [INFO] Setting property: resource.loader => 'classpath'.
> [INFO] Setting property: resource.manager.logwhenfound => 'false'.
> [WARNING] No archetype repository found. Falling back to central repository (http://repo1.maven.org/maven2).
> [WARNING] Use -DarchetypeRepository=<your repository> if archetype's repository is elsewhere.
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD FAILURE
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time:  0.554 s
> [INFO] Finished at: 2023-01-19T10:38:11-06:00
> [INFO] ------------------------------------------------------------------------
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-archetype-plugin:2.0-alpha-2:generate (default-cli) on project standalone-pom: The defined artifact is not an archetype -> [Help 1]
> [ERROR]
> [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
> [ERROR] Re-run Maven using the -X switch to enable full debug logging.
> [ERROR]
> [ERROR] For more information about the errors and possible solutions, please read the following articles:
> [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

According to the stack trace, the cause is the defined artificat is not an archetype

> Failed to execute goal org.apache.maven.plugins:maven-archetype-plugin:2.0-alpha-2:generate (default-cli) on project standalone-pom: The defined artifact is not an archetype -> [Help 1]

>

**Reply by joschi** [reply]
Check your Maven setup.

This is working for me:
```
# mvn -v
Apache Maven 3.8.7 (b89d5959fcde851dcb1c8946a785a163f14e1e29)
Maven home: /path/to/maven/3.8.7/libexec
Java version: 19.0.1, vendor: Homebrew, runtime: /path/to/java/19.0.1/libexec/openjdk.jdk/Contents/Home
Default locale: en_DE, platform encoding: UTF-8
OS name: "mac os x", version: "12.6.2", arch: "aarch64", family: "mac"

# mvn archetype:generate -DarchetypeGroupId=io.dropwizard.archetypes -DarchetypeArtifactId=java-simple -DarchetypeVersion=2.1.4
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------< org.apache.maven:standalone-pom >-------------------
[INFO] Building Maven Stub Project (No POM) 1
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] >>> maven-archetype-plugin:3.2.1:generate (default-cli) > generate-sources @ standalone-pom >>>
[INFO] 
[INFO] <<< maven-archetype-plugin:3.2.1:generate (default-cli) < generate-sources @ standalone-pom <<<
[INFO] 
[INFO] 
[INFO] --- maven-archetype-plugin:3.2.1:generate (default-cli) @ standalone-pom ---
[INFO] Generating project in Interactive mode
[INFO] Archetype repository not defined. Using the one from [io.dropwizard.archetypes:java-simple:4.0.0-beta.3] found in catalog remote
Define value for property 'name': MyProject
[INFO] Using property: description = null
[INFO] Using property: shaded = true
Define value for property 'groupId': com.example.dropwizard
Define value for property 'artifactId': my-project  
Define value for property 'version' 1.0-SNAPSHOT: : 
Define value for property 'package' com.example.dropwizard: : 
Confirm properties configuration:
name: MyProject
description: null
shaded: true
groupId: com.example.dropwizard
artifactId: my-project
version: 1.0-SNAPSHOT
package: com.example.dropwizard
 Y: : y
[INFO] ----------------------------------------------------------------------------
[INFO] Using following parameters for creating project from Archetype: java-simple:2.1.4
[INFO] ----------------------------------------------------------------------------
[INFO] Parameter: groupId, Value: com.example.dropwizard
[INFO] Parameter: artifactId, Value: my-project
[INFO] Parameter: version, Value: 1.0-SNAPSHOT
[INFO] Parameter: package, Value: com.example.dropwizard
[INFO] Parameter: packageInPathFormat, Value: com/example/dropwizard
[INFO] Parameter: shaded, Value: true
[INFO] Parameter: package, Value: com.example.dropwizard
[INFO] Parameter: groupId, Value: com.example.dropwizard
[INFO] Parameter: name, Value: MyProject
[INFO] Parameter: description, Value: null
[INFO] Parameter: artifactId, Value: my-project
[INFO] Parameter: version, Value: 1.0-SNAPSHOT
[INFO] Project created from Archetype in dir: /path/to/my-project
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  25.552 s
[INFO] Finished at: 2023-01-20T09:17:47+01:00
[INFO] ------------------------------------------------------------------------
```

**Reply by DavidNovo** [reply]
Here is my set-up
─(19:00:09)──> mvn -v                          1 ↵ ──(Sun,Jan22)─┘
Apache Maven 3.8.7 (b89d5959fcde851dcb1c8946a785a163f14e1e29)
Maven home: /usr/local/Cellar/maven/3.8.7/libexec
Java version: 18.0.2, vendor: Amazon.com Inc., runtime: /Users/davidnovogrodsky_wrk/.sdkman/candidates/java/18.0.2-amzn
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "12.6.2", arch: "x86_64", family: "mac"

And here is the result:

> └─(19:03:05)──> mvn archetype:generate -DarchetypeGroupId=io.dropwizard.archetypes -DarchetypeArtifactId=java-simple -DarchetypeVersion=2.1.4
> [INFO] Scanning for projects...
> [INFO]
> [INFO] ------------------< org.apache.maven:standalone-pom >-------------------
> [INFO] Building Maven Stub Project (No POM) 1
> [INFO] --------------------------------[ pom ]---------------------------------
> [INFO]
> [INFO] >>> maven-archetype-plugin:2.0-alpha-2:generate (default-cli) > generate-sources @ standalone-pom >>>
> [INFO]
> [INFO] <<< maven-archetype-plugin:2.0-alpha-2:generate (default-cli) < generate-sources @ standalone-pom <<<
> [INFO]
> [INFO]
> [INFO] --- maven-archetype-plugin:2.0-alpha-2:generate (default-cli) @ standalone-pom ---
> [INFO] Setting property: classpath.resource.loader.class => 'org.codehaus.plexus.velocity.ContextClassLoaderResourceLoader'.
> [INFO] Setting property: velocimacro.messages.on => 'false'.
> [INFO] Setting property: resource.loader => 'classpath'.
> [INFO] Setting property: resource.manager.logwhenfound => 'false'.
> [WARNING] No archetype repository found. Falling back to central repository (http://repo1.maven.org/maven2).
> [WARNING] Use -DarchetypeRepository=<your repository> if archetype's repository is elsewhere.
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD FAILURE
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time:  0.634 s
> [INFO] Finished at: 2023-01-22T19:03:41-06:00
> [INFO] ------------------------------------------------------------------------
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-archetype-plugin:2.0-alpha-2:generate (default-cli) on project standalone-pom: The defined artifact is not an archetype -> [Help 1]
> [ERROR]
> [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
> [ERROR] Re-run Maven using the -X switch to enable full debug logging.
> [ERROR]
> [ERROR] For more information about the errors and possible solutions, please read the following articles:
> [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

And here is my settings.xml file
`<?xml version="1.0" encoding="UTF-8"?>
<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 https://maven.apache.org/xsd/settings-1.0.0.xsd">

</settings>`
The settings.xml file has no information in it.  Do I need to add stuff to this file?

**Reply by DavidNovo** [reply]
Also, than you so much r your help.

**Reply by DavidNovo** [reply]
I found the problem.  I had to add a -U flag to the end of the command to force Maven to update the local repo with the new archetypes.

thanks again for your help.
